### PR TITLE
[t2251] weighted zooming around scrubber

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/8f7748682875785/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/8f7748682875785/output.txt

Total script time: 3.00 mins

File: src/timeline/super-scrollbar.js [hunk 1]
```diff
@@ -241,20 +254,50 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
+        // Otherwise, derive a ratio representing the distance from the viewport's left side to the
+        // scrubber's position with respect to the width of the viewport.
+        p = ( scrubberRect.left - viewportRect.left ) / viewportRect.width;
+
+        // Assign oppositely-weighted values to left and right growth variables.
+        leftGrowth = widthGrowth * p;
+        rightGrowth = widthGrowth * ( 1 - p );
+      }
+
+      // Come up with new pixel values for left, right and width of viewport.
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js:288]: zoomIn does not have this comment.

File: src/timeline/super-scrollbar.js [hunk 2]
```diff
@@ -241,20 +254,50 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
+      leftPosition = _viewPort.offsetLeft - leftGrowth;
+      rightPosition = _rect.right - viewportRect.right - rightGrowth;
+      newWidth = viewportRect.width + widthGrowth;
 
+      // Find the ratio of left and right position width respect to the super-scrollbar's width.
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js:293]: same as above.

File: src/timeline/super-scrollbar.js [hunk 3]
```diff
@@ -241,20 +254,50 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
       leftPosition = ( leftPosition < 0 ? 0 : leftPosition ) / _element.clientWidth;
-      rightPosition = ( rightPosition < 0 ? 0 : rightPosition) / _element.clientWidth;
+      rightPosition = ( rightPosition < 0 ? 0 : rightPosition ) / _element.clientWidth;
 
       // Proxy for applying viewport-transition so it happens safely and gets removed properly.
       refreshTransitionLock();
 
+      // Assign percentage values to viewport style (for most accuracy).
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js:300]: Same as above.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/bcc9c2eb3a304c8/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/bcc9c2eb3a304c8/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7c2666ad910bdf4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/7c2666ad910bdf4/output.txt

Total script time: 1.58 mins

Server is running on http://pr1025.butyr.org/templates/basic/
Run the Test Suite: http://pr1025.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b081aa6a41b7c4a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/b081aa6a41b7c4a/output.txt

Total script time: 2.30 mins

Server is running on http://pr1025.butyr.org/templates/basic/
Run the Test Suite: http://pr1025.butyr.org/test/

**Pull request closed without merging**